src/rdkafka_cgrp.c

```diff
@@ -5098,6 +5098,49 @@ void rd_kafka_cgrp_set_member_id (rd_kafka_cgrp_t *rkcg, const char *member_id){
 }
 
 
+/**
+ * @brief Determine owned partitions that no longer exist (partitions in
+ *        deleted topics).
+ */
+static rd_kafka_topic_partition_list_t *
+rd_kafka_cgrp_owned_but_not_exist_partitions(rd_kafka_cgrp_t *rkcg) {
+        rd_kafka_topic_partition_list_t *result = NULL;
+        int i, j;
+
+        // TODO: something that isn't O(N*M)
+        for (i=0; i<rkcg->rkcg_assignment->size; i++) {
+                rd_bool_t found = rd_false;
+                const rd_kafka_topic_partition_t *cur =
+                        &rkcg->rkcg_assignment->elems[i];
+
+                for (j=0; j<rkcg->rkcg_subscribed_topics->rl_cnt; j++) {
+                        const char *topic = ((rd_kafka_topic_info_t *)
+                                rkcg->rkcg_subscribed_topics->rl_elems[j])->topic;
+
+                        if (!strcmp(topic, cur->topic)) {
+                                found = rd_true;
+                                break;
+                        }
+                }
+
+                if (found)
+                        continue;
+
+                if (!result)
+                        result = rd_kafka_topic_partition_list_new(
+                                        rkcg->rkcg_assignment->cnt);
+
+                rd_kafka_topic_partition_list_add0(result,
+                                                   cur->topic,
+                                                   cur->partition,
+                                                   cur->_private);
```

Comment:
rktpar->_private is a rktp reference, which needs to be refcounted.
So do a: `list_add0(..., cur->_private ? rd_kafka_toppar_keep(cur->_private) : NULL)` or simply use `rd_kafka_topic_partition_list_add_copy()` which already does this.

Reply:
`rd_kafka_topic_partition_list_add0` does refcount the _private. the problem is actually occurring without any of the additional code i've added in this method, on another thread i think. feel free to have another look - i may be one step away from working this out, or i may not.